src/creature.cpp

```diff
@@ -246,27 +246,33 @@ bool Creature::getNextStep(Direction& dir, uint32_t&)
 		return false;
 	}
 
-	dir = listWalkDir.front();
-	listWalkDir.pop_front();
+	dir = listWalkDir.back();
+	listWalkDir.pop_back();
 	onWalk(dir);
 	return true;
 }
 
-void Creature::startAutoWalk(const std::forward_list<Direction>& listDir)
+void Creature::startAutoWalk()
 {
 	Player* player = getPlayer();
 	if (player && player->isMovementBlocked()) {
 		player->sendCancelWalk();
 		return;
 	}
 
-	listWalkDir = listDir;
+	addEventWalk(listWalkDir.size() > 0);
+}
 
-	size_t size = 0;
-	for (auto it = listDir.begin(); it != listDir.end() && size <= 1; ++it) {
-		size++;
+void Creature::startAutoWalk(const std::vector<Direction>& listDir)
+{
+	Player* player = getPlayer();
+	if (player && player->isMovementBlocked()) {
+		player->sendCancelWalk();
+		return;
 	}
-	addEventWalk(size == 1);
+
+	listWalkDir.assign(listDir.begin(), listDir.end());
+	addEventWalk(listWalkDir.size() > 0);
```

Comment:
```suggestion
	addEventWalk(!listWalkDir.empty());
```

Reply:
Fixed. To keep the old behavior(before this pr) it should be addEventWalk(listWalkDir.size() == 1);